### PR TITLE
*: introduce the non-retryable error FlashbackInProgress

### LINKED ISSUE #13397: Introduce the non-retryable error FlashbackInProgress
### Is your feature request related to a problem? Please describe:

For now, if a read/write request is sent to a region in the flashback progress, it will reuse the `RecoveryInProgress` error to indicate that this region is temporarily inaccessible. However, this error is [set to retryable](https://github.com/tikv/client-go/pull/485) in the client-go, which is somewhat incompatible with the flashback's feature. Since we expect that the workload should be stopped during the flashback progress, if a request meets the `FlashbackInProgress` error, it should stop retrying immediately to avoid unnecessary backoff and potential unexpected data status to the user.

### Describe the feature you'd like:

Return the non-retryable error `FlashbackInProgress` rather than `RecoveryInProgress` during the flashback progress.

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: Close #13397. Should merge https://github.com/pingcap/kvproto/pull/982 before this PR.

What's Changed:

```commit-message
Introduce the non-retryable error `FlashbackInProgress` for the region in the flashback progress to reject any read or write.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by HuSharp — APPROVED** [review]
LGTM

File: etc/error_code.toml [hunk 1]
```diff
@@ -1,11 +1,11 @@
-["KV:Cloud:IO"]
+["KV:Cloud:Io"]
```
Comment (Connor1996) [inline etc/error_code.toml:1]: why change

**Review by Connor1996 — COMMENTED** [review]
should update the client side first

File: etc/error_code.toml (hunk 1 above)
Comment (JmPotato) [reply etc/error_code.toml:1]: https://github.com/tikv/tikv/pull/13215 changed all namings to use the camel case but without calling `make error-code`, which I did.

Comment (Connor1996) [reply etc/error_code.toml:1]: That's intentional. Not sure if the error code is of case sensitivity. /cc @5kbpers

Comment (5kbpers) [reply etc/error_code.toml:1]: It's OK because error codes are used for printing only currently.

**Review by 5kbpers — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**